lib/utils.js

```diff
@@ -126,10 +127,32 @@ function urlFromContext (context) {
  * @return {Object}
  */
 function getModelFromContext (context, app) {
+  var type = getTypeFromContext(context)
+  if (app.models[type]) return app.models[type]
+
   var name = modelNameFromContext(context)
   return app.models[name]
 }
 
+/**
+ * Returns a model type from the context object.
+ * Infer the type from the `root` returns in the remote.
+ * @public
+ * @memberOf {Utils}
+ * @param {Object} context
+ * @param {Object} app
+ * @return {String}
+ */
+function getTypeFromContext (context) {
+  if (!context.method.returns) return undefined
+
+  const returns = [].concat(context.method.returns)
+  for (var i = 0, l = returns.length; i < l; i++) {
+    if (typeof returns[i] !== 'object' || returns[i].root !== true) continue
+    return returns[i].type
```

Comment:
should you check if `returns[i]` has a type? Or is it fine that it will return `undefined` if no type is specified?

Reply:
Perfectly fine (like line 147 I `return undefined`, cause no types are availble. I'm making a diferenciation between `type` and `modelName` in this method. Then the `getModelFromContext` choose which one to use.